pkg/apply/event/event.go

```diff
@@ -75,92 +81,73 @@ type ErrorEvent struct {
 	Err error
 }
 
-//go:generate stringer -type=ApplyEventType
-type ApplyEventType int
+//go:generate stringer -type=ActionGroupEventType
+type ActionGroupEventType int
 
 const (
-	ApplyEventResourceUpdate ApplyEventType = iota
-	ApplyEventCompleted
+	Started ActionGroupEventType = iota
+	Finished
 )
 
+type ActionGroupEvent struct {
+	GroupName string
+	Action    ResourceAction
+	Type      ActionGroupEventType
+}
+
 //go:generate stringer -type=ApplyEventOperation
 type ApplyEventOperation int
 
 const (
-	ServersideApplied ApplyEventOperation = iota
+	NoneApply ApplyEventOperation = iota
```

Comment:
The `Failed` operation is removed. How does the event consumer know the apply is failed? Is it by checking the `Error` field in the ApplyEvent?

Reply:
Yeah, the idea is to avoid having two places where users can check for errors/failures. It is not clear what it means if an ApplyEvent has operation `Failed` but no error, or the other way around. Do we have examples where having a separate `operation` would be useful?

Reply:
I'm not aware of such examples.